src/storage/heap_file.c

```diff
@@ -22586,6 +22551,17 @@ heap_update_home (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * context, boo
   assert (context->home_page_watcher_p->pgptr != NULL);
   assert (context->forward_page_watcher_p != NULL);
 
+   // *INDENT-OFF*
+   // To ensure that, if started, the atomic replication area will also end.
+   scope_exit <std::function<void (void)>> log_on_exit ([&thread_p, &atomic_replication_flag]()
+   {
+     if (atomic_replication_flag == true)
+      {
+         log_append_empty_record (thread_p, LOG_END_ATOMIC_REPL, NULL);
+      }
+   });
+   // *INDENT-ON*
+
```

Comment:
1. LOG_START_ATOMIC_REPL is never left in heap_update_home().
2. I think we must leave them some cases..

Reply:
Oh, it is left when `is_mvcc_op` is true. My bad, again, sorry.

Reply:
I guess we can make change to 22711L
`if (!atomic_replication_flag && is_mvcc_op)`
->
`if (is_mvcc_op)`
And I think we can get rid of `atomic_replication_flag` and just use only `is_mvcc_op`.

Reply:
My opinion is to keep the `_flag` - it's the same elsewhere.
It's just more explicit even if redundant.

Nit.
Using only the `is_mvcc_op` argument is actually prone to bugs.
Since `bool is_mvcc_op` is not `const`, it might be modified within the function (even if accidentally) - so, by the end of the function, when it would be used in the lambda to add the closing log record it's value might not be the same.
I would make that argument `const` but this is outside the scope of the PR.